Yoast/YoastSEO.js
# Add Spanish participle precedence exception, improve French and English word lists

## Summary

This PR can be summarized in the following changelog entry:
* Adds a filter to mark Spanish sentences as non-passive when certain exception words occur between the auxiliary and the participle. The list of exception words includes all forms of the copula _estar_.


In addition, some changes were made to various word lists:

1. Some categories were added to the filter that marks Spanish sentences as non-passive  when certain exception words directly precede the participle:
    - interrogativeDeterminers
    - interrogativePronouns
    - personalPronounsComitative
    - personalPronounsPrepositional
    - prepositionalAdverbs

2. The following words were added to the French function words:
additional French function words:
    - articles: “l’un”, “l’une”
    - cardinalNumerals: “septante”, “huitante”, “octante”, “nonante”
    - personalPronounsNominative: “qu’il”, “qu’elle”, “qu’ils”, “qu’elles”, “qu’on”, “d’il”, “d’elle”, “d’ils”, “d’elles”
    - personalPronounsStressed: “d’eux”, “qu’eux”
    - indefinitePronouns: “d’autres”
    - pronominalAdverbs: “n’y”
    - locativeAdverbs: “d’ici”
    - otherAuxiliaries: “n’a”, “n’ai”, “n’aie”, “n’as”,  “n’aura", “n’aurai", “n’auraient", “n’aurais", “n’aurait", “n’auras", “n’aurez", “n’auriez", “n’aurons", “n’auront", “n’avaient", “n’avais", “n’avait", “n’avez", “n’avez-vous", “n’aviez", “n’avions", “n’avons", “n’avons-nous", “n’ayez", “n’ayons", “n’ont", “n’ont-elles", “n’ont-ils", “n’allons", “n’allez", “n’allais", “n’allait", “n’allions", “n’alliez", “n’allaient", “n’iras", “n’ira", “n’irons", “n’irez", “n’iront", “qu’a”
    - otherAuxiliariesInfinitive: “d’avoir”, “d’aller”, “n’avoir”
    - copula: “n’es”, “n’est-ce”, “n’êtes”, “n’était”, “n’étais”, “n’étions”, “n’étiez”, “n’étaient”, “qu’est”
    - copulaInfinitive: “d’être”
    - prepositions:  “d’après”

3. The following words were deleted from the French irregular participles list:
    descendu, échu, reparti, redescendu, ressorti, sorti

4. The following words were added to the French non-participles ending with -é:
    débellé, ressuscité, retourné, remonté, rentré, monté, passé

5. The following word was added to the English non-verbs ending in -ed:
    flawed

## Test instructions

This PR can be tested by following these steps:

- Check out the branch.
- Use the browserified example. Don't forget to run a `grunt build:js`.
-  Set the locale to `es_ES`.
- Make sure the following sentence is not marked as passive:
Los niños que son famosos estaban en un programa de televisión muy esperado.
- The changes to the word lists do not require any additional testing.

Fixes #1387
